exporter/exporterhelper/request.go

```diff
@@ -0,0 +1,50 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package exporterhelper // import "go.opentelemetry.io/collector/exporter/exporterhelper"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/exporter/exporterhelper/internal"
+)
+
+// Request represents a single request that can be sent to an external endpoint.
+type Request interface {
+	// Export exports the request to an external endpoint.
+	Export(ctx context.Context) error
+}
+
+// RequestItemsCounter is an optional interface that can be implemented by Request to provide a number of items
+// in the request. This is a recommended interface to implement for exporters. It is required for batching and queueing
+// based on number of items. Also, it's used for reporting number of items in collector's logs, metrics and traces.
+// If not implemented, collector's logs, metrics and traces will report 0 items.
+type RequestItemsCounter interface {
+	// ItemsCount returns a number of basic items in the request where item is the smallest piece of data that can be
+	// sent. For example, for OTLP exporter, this value represents the number of spans,
+	// metric data points or log records.
+	ItemsCount() int
+}
+
+type request struct {
+	Request
+	baseRequest
+}
+
+var _ internal.Request = (*request)(nil)
+
+func (req *request) OnError(_ error) internal.Request {
```

Comment:
are OnError and Count defined in an interface somewhere?

Reply:
Yes, they are part of the internal.Request interface https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/exporterhelper/internal/request.go

Reply:
Ah yes... this was the piece i was missing 👍🏻